internal/core/commands/sync_dd_power_train_type.go

```diff
@@ -0,0 +1,201 @@
+	return SyncPowerTrainTypeCommandHandler{DBS: dbs, logger: logger}
+}
+
+func (ch SyncPowerTrainTypeCommandHandler) Handle(ctx context.Context, _ mediator.Message) (interface{}, error) {
+
+	all, err := models.DeviceDefinitions(models.DeviceDefinitionWhere.Verified.EQ(true),
+		qm.Load(models.DeviceDefinitionRels.DeviceStyles),
+		qm.Load(models.DeviceDefinitionRels.DeviceType),
+		qm.Load(models.DeviceDefinitionRels.DeviceMake)).All(ctx, ch.DBS().Reader)
+	if err != nil {
+		return nil, err
+	}
+
+	ch.logger.Info().Msgf("found %d device definitions verified, starting process...", len(all))
+	if len(all) == 0 {
+		return SyncPowerTrainTypeCommandResult{false}, nil
+	}
+
+	content, err := os.ReadFile("powertrain_type_rule.yaml")
+	if err != nil {
+		return nil, err
+	}
+
+	var powerTrainTypeData coremodels.PowerTrainTypeRuleData
+	err = yaml.Unmarshal(content, &powerTrainTypeData)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, definition := range all {
+		ch.logger.Info().Msgf("%s - Make:%s Model: %s Year: %d", definition.ID, definition.R.DeviceMake.NameSlug, definition.ModelSlug, definition.Year)
+
+		if definition.R.DeviceType == nil {
+			ch.logger.Error().Msgf("ID: %s with DeviceType is empty", definition.ID)
+			continue
+		}
+
+		metadataKey := definition.R.DeviceType.Metadatakey
+		var metadataAttributes map[string]any
+
+		if err := definition.Metadata.Unmarshal(&metadataAttributes); err == nil {
+			metadataAttributes = make(map[string]interface{})
+			metaData := make(map[string]interface{})
+
+			dt, err := models.DeviceTypes(models.DeviceTypeWhere.ID.EQ(definition.DeviceTypeID.String)).
+				One(ctx, ch.DBS().Reader)
+
+			if err != nil {
+				if !errors.Is(err, sql.ErrNoRows) {
+					return nil, err
+				}
+			}
+
+			var deviceTypeAttributes map[string][]coremodels.GetDeviceTypeAttributeQueryResult
+			if err := dt.Properties.Unmarshal(&deviceTypeAttributes); err == nil {
+				for _, deviceAttribute := range deviceTypeAttributes["properties"] {
+					metaData[deviceAttribute.Name] = deviceAttribute.DefaultValue
+				}
+			}
+
+			metadataAttributes[metadataKey] = metaData
+		}
+
+		for key, _ := range metadataAttributes[metadataKey].(map[string]any) {
+			if key == "powertrain_type" {
+				powerTrainType := ch.resolvePowerTrainTypeByMake(ctx, powerTrainTypeData, definition)
+				metadataAttributes[metadataKey].(map[string]interface{})["powertrain_type"] = powerTrainType
+			}
+		}
+
+		err = definition.Metadata.Marshal(metadataAttributes)
+		if err != nil {
+			return nil, err
+		}
+
+		if err = definition.Upsert(ctx, ch.DBS().Writer, true, []string{models.DeviceDefinitionColumns.ID}, boil.Infer(), boil.Infer()); err != nil {
+			return nil, err
+		}
+	}
+
+	return SyncPowerTrainTypeCommandResult{true}, nil
+}
+
+func (ch SyncPowerTrainTypeCommandHandler) resolvePowerTrainTypeByMake(ctx context.Context, powerTrainTypeData coremodels.PowerTrainTypeRuleData,
```

Comment:
```suggestion
func (ch SyncPowerTrainTypeCommandHandler) resolvePowerTrainType(ctx context.Context, powerTrainTypeData coremodels.PowerTrainTypeRuleData,
```
Consider renaming since this function does more than just resolving By Make